### PR TITLE
Support b3 http propagation format for jaeger

### LINKED ISSUE #354: Support zipkin b3 http propagation format for jaeger
Jaeger users should be able to set b3 http propagation format

### PR DESCRIPTION
**Pull request by rafaeljesus**
Support zipkin b3 http propagation format for jaeger. Defaults to jaeger propagation mechanism if format is not defined.

closes https://github.com/hellofresh/janus/issues/354

### CODE REVIEW

File: pkg/config/specification.go [hunk 1]
```diff
@@ -120,6 +120,7 @@ type JaegerTracing struct {
 	BufferFlushInterval time.Duration `envconfig:"TRACING_JAEGER_BUFFER_FLUSH_INTERVAL"`
 	LogSpans            bool          `envconfig:"TRACING_JAEGER_LOG_SPANS"`
 	QueueSize           int           `envconfig:"TRACING_JAEGER_QUEUE_SIZE"`
+	PropagationFormat   string        `envconfig:"TRACING_JAEGER_PROPAGATION_FORMAT"`
```
Comment (vgarvardt) [inline pkg/config/specification.go:123]: this must be documented somewhere

File: pkg/opentracing/opentracing.go [hunk 2]
```diff
@@ -95,11 +97,43 @@ func (t *Tracing) buildJaeger(componentName string, c config.JaegerTracing) (ope
...
 
-	return cfg.New(
-		componentName,
-		jaegercfg.Logger(jaegerLoggerAdapter{log.StandardLogger()}),
-		jaegercfg.Metrics(metrics.NullFactory),
+	tracerMetrics := jaeger.NewMetrics(metrics.NullFactory, nil)
+	tracerLogger := jaegerLoggerAdapter{log.StandardLogger()}
+	sampler, err := cfg.Sampler.NewSampler(componentName, tracerMetrics)
+	if err != nil {
+		log.WithError(err).Error("failed to create jaeger sampler")
```
Comment (vgarvardt) [inline pkg/opentracing/opentracing.go]: `sampler` creation failed, but you're not returning from the function and trying to use broken sampler bellow

File: pkg/opentracing/opentracing.go [hunk 3]
```diff
@@ -95,11 +97,43 @@ func (t *Tracing) buildJaeger(componentName string, c config.JaegerTracing) (ope
...
+		log.WithError(err).Error("failed to create jaeger reporter")
+	}
+
+	var (
+		tracer opentracing.Tracer
+		closer io.Closer
 	)
+
+	switch c.PropagationFormat {
+	case "zipkin":
```
Comment (vgarvardt) [inline pkg/opentracing/opentracing.go]: `const` pls

File: pkg/opentracing/opentracing.go [hunk 4]
```diff
@@ -11,13 +11,13 @@ import (
 	log "github.com/sirupsen/logrus"
 	jaeger "github.com/uber/jaeger-client-go"
 	jaegercfg "github.com/uber/jaeger-client-go/config"
-	"github.com/uber/jaeger-client-go/zipkin"
```
Comment (vgarvardt) [inline pkg/opentracing/opentracing.go]: looks like an issue =)

Comment (rafaeljesus) [reply pkg/opentracing/opentracing.go]: opsss

**Review by vgarvardt — APPROVED** [review]

**Pull request merged**